pkg/kwokctl/runtime/binary/cluster.go

```diff
@@ -543,36 +544,78 @@ func (c *Cluster) Down(ctx context.Context) error {
 	return nil
 }
 
-func (c *Cluster) getComponent(ctx context.Context, name string) (internalversion.Component, error) {
+func (c *Cluster) Start(ctx context.Context) error {
 	config, err := c.Config(ctx)
 	if err != nil {
-		return internalversion.Component{}, err
+		return err
 	}
-	component, ok := slices.Find(config.Components, func(component internalversion.Component) bool {
-		return component.Name == name
-	})
-	if !ok {
-		return internalversion.Component{}, fmt.Errorf("%w: %s", runtime.ErrComponentNotFound, name)
+
+	err = c.startComponents(ctx, config.Components)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Cluster) Stop(ctx context.Context) error {
+	config, err := c.Config(ctx)
+	if err != nil {
+		return err
 	}
 
-	return component, nil
+	err = c.stopComponents(ctx, config.Components)
+	if err != nil {
+		return err
+	}
+
+	return nil
 }
 
-func (c *Cluster) Start(ctx context.Context, name string) error {
-	component, err := c.getComponent(ctx, name)
+func (c *Cluster) StartComponent(ctx context.Context, name string) error {
+	conf, err := c.Config(ctx)
 	if err != nil {
 		return err
 	}
 
-	err = c.startComponent(ctx, component)
+	_, err = c.GetComponent(ctx, name)
+	if err != nil {
+		if !errors.Is(err, runtime.ErrComponentNotFound) {
+			return err
+		}
+	} else {
+		return nil
+	}
+
+	err = c.startComponent(ctx, internalversion.Component{
+		Name: name,
+	})
```

Comment:
```suggestion
	component, err = c.GetComponent(ctx, name)
	if err != nil {
		return err
	}

	err = c.startComponent(ctx, component)
```

This configuration only indicates which components are in the cluster, and does not include their running status, so perhaps we could add a `status.components` if necessary.